### PR TITLE
add support for OTLP headers from environment

### PR DESCRIPTION
**Pull request by spookyvision**
This PR adds support for reading `key1=value1,key2=value2,...` tuples from `OTEL_EXPORTER_OTLP_HEADERS` and `OTEL_EXPORTER_OTLP_TRACES_HEADERS`, thus enabling a smoother quick start from [honeycomb's setup guide](https://docs.honeycomb.io/getting-data-in/opentelemetry-overview/).

### CODE REVIEW

File: init-tracing-opentelemetry/src/tracing_subscriber_ext.rs [hunk 1]
```diff
@@ -76,7 +75,7 @@ where
     // )?;
     init_propagator()?;
     Ok(tracing_opentelemetry::layer()
-        .with_exception_field_propagation(true)
+        .with_error_records_to_exceptions(true)
```
Comment (davidB) [inline init-tracing-opentelemetry/src/tracing_subscriber_ext.rs:78]: Why did you change it?
What is the effect?

**Review by davidB — COMMENTED** [review]
Thanks for the PR

File: init-tracing-opentelemetry/src/tracing_subscriber_ext.rs (hunk 1 above)
Comment (spookyvision) [reply init-tracing-opentelemetry/src/tracing_subscriber_ext.rs:78]: the other method was marked as deprecated
